test/executor/test_sample_executor.py

```diff
@@ -0,0 +1,38 @@
+import unittest
+import pandas as pd
+import numpy as np
+import math
+
+from src.executor.sample_executor import SampleExecutor
+from src.models.storage.batch import Batch
+from test.executor.utils import DummyExecutor
+from src.planner.sample_plan import SamplePlan
+from src.expression.constant_value_expression import ConstantValueExpression
+
+
+class SampleExecutorTest(unittest.TestCase):
+
+    def test_should_return_smaller_num_rows(self):
+        dfs = [pd.DataFrame(np.random.randint(0, 100, size=(100, 4)),
+                            columns=list('ABCD')) for _ in range(4)]
+
+        batches = [Batch(frames=df) for df in dfs]
+
+        sample_value = 5
+
+        target_size = 0
+        for batch in batches:
+            target_size += batch.batch_size
+        target_size = int(math.ceil(target_size / sample_value))
+
+        plan = SamplePlan(ConstantValueExpression(sample_value))
+
+        sample_executor = SampleExecutor(plan)
+        sample_executor.append_child(DummyExecutor(batches))
+        reduced_batches = list(sample_executor.exec())
+
+        total_size = 0
+        for batch in reduced_batches:
+            total_size += batch.batch_size
+
+        self.assertEqual(total_size, target_size)
```

Comment:
Compare the content and use sample_value that does not divides the batch_size.

Reply:
Fixed.